owtf/api/handlers/auth.py

```diff
@@ -0,0 +1,230 @@
+        """
+            Set user cookie for the logged-in user
+            Setting HttpOnly attribute for preventing XSS attacks
+            Setting SameSite for preventing CSRF Attacks    
+        """
+
+        self.application.settings['cookie_secret'] = base64.b64encode(
+            uuid.uuid4().bytes + uuid.uuid4().bytes).decode('utf-8')
+        if user:
+            self.set_secure_cookie("user", tornado.escape.json_encode(user), secure=True, HttpOnly=True, SameSite='Lax')
+        else:
+            self.clear_cookie("user")
+
+
+class RegisterHandler(APIRequestHandler):
+    """Registers a new user when he provides email, name, password and confirm password"""
+
+    SUPPORTED_METHODS = ["POST"]
+
+    def post(self):
+        """
+            **Example request**:
+
+            .. sourcecode:: http
+
+            POST /api/v1/register/ HTTP/1.1
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "email": "test@test.com",
+                "password": "Test@34335",
+                "confirm_password": "Test@34335",
+                "name": "test"
+            }
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            **Successful registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok"
+                }
+            }
+
+            **Failed registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "fail",
+                    "message": "Email already exists"
+                }
+            }
+
+        """
+        body_data = json.loads(self.request.body.decode('utf-8'))
+        name = body_data['name']
+        email = body_data['email']
+        password = body_data['password']
+        confirm_password = body_data['confirm_password']
+
+        already_taken = User.find_by_email(self.session, email)
+        reg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{8,20}$"
+        pat = re.compile(reg)
+        mat = re.search(pat, password)
+        data = {
+            "status": "fail",
+        }
+        if password != confirm_password:
+            data["message"] = "Password doesn't match"
+            self.success(data)
+        elif not mat:
+            data["message"] = "Choose a strong password"
+            self.success(data)
+        elif already_taken:
+            data["message"] = "Email already exists"
+            self.success(data)
+        else:
+            hashed_pass = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt(8))
```

Comment:
why only a 8 byte salt? the correct usage is to use the default `bcrypt.gensalt()`

Reply:
Corrected.